monitoring-6.x.md

```diff
@@ -0,0 +1,948 @@
+   </td>
+   <td>Error when systemd detects a failed service
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >Filesystem
+   </td>
+   <td>High disk space usage
+   </td>
+   <td>Warning at > 80% used
+<p>
+Critical error at > 90% used
+   </td>
+  </tr>
+  <tr>
+   <td>High inode usage
+   </td>
+   <td>Warning at > 90% used
+<p>
+Critical error at > 95% used
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >System
+   </td>
+   <td>Uptime
+   </td>
+   <td>Warning node uptime &lt; 5 mins
+   </td>
+  </tr>
+  <tr>
+   <td>Kernel params
+   </td>
+   <td>Error if param not set
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >Etcd
+   </td>
+   <td>Etcd instance health
+   </td>
+   <td>Error when etcd master down > 5 mins
+   </td>
+  </tr>
+  <tr>
+   <td>Etcd latency check
+   </td>
+   <td>Warning when follower &lt;-> leader latency > 500 ms
+<p>
+Error when > 1 sec over period of 1 min
+   </td>
+  </tr>
+  <tr>
+   <td>Docker
+   </td>
+   <td>Docker daemon health
+   </td>
+   <td>Error when docker daemon is down
+   </td>
+  </tr>
+  <tr>
+   <td>Kubernetes
+   </td>
+   <td>Kubernetes node readiness
+   </td>
+   <td>Error when the node is not ready
+   </td>
+  </tr>
+</table>
+
+### Alertmanager Email Configuration
+
+In order to configure email alerts via Alertmanager you will need to create Gravity resources of type `smtp `and `alerttarget`.
+
+An example of the configuration is shown below:
+
+```
+kind: smtp
+version: v2
+metadata:
+  name: smtp
+spec:
+  host: smtp.host
+  port: <smtp port> # 465 by default
+  username: <username>
+  password: <password>
+---
+kind: alerttarget
+version: v2
+metadata:
+  name: email-alerts
+spec:
+  email: triage@example.com # Email address of the alert recipient
+```
+
+Creating these resources will accordingly update and reload Alertmanager configuration:
+
+```
+$ gravity resource create -f smtp.yaml
+```
+
+In order to view the current SMTP settings or alert target:
+
+```
+$ gravity resource get smtp
+$ gravity resource get alerttarget
+```
+
+Only a single alert target can be configured. To remove the current alert target, you can execute the following command:
+
+```
+$ gravity resource rm alerttarget email-alerts
+```
+
+### Testing Alertmanager Email Configuration
+
+To test a Alertmanager SMTP configuration you can execute the following:
+
+```
+$ kubectl exec -n monitoring $POD_ID -c alertmanager -- /bin/bash -c "alertmanager service-tests smtp"
+```
+
+If the settings are set up appropriately, the recipient should receive an email with the subject “test subject”.
+
+### Alertmanager Custom Alerts
+
+Creating new alerts is as easy as using another Gravity resource of type `alert`. The alerts are written in TICKscript and are automatically detected, loaded, and enabled for Gravity Monitoring and Alerts system.
+
+For demonstration purposes let’s define an alert that always fires:
+
+```
+kind: alert
+version: v2
+metadata:
+  name: my-formula
+spec:
+  formula: |
```

Comment:
This is Kapacitor's alert definition, it won't work with Alertmanager.

Reply:
I have an example of a custom Alertmanager alert that always fires:

```yaml
kind: alert
version: v2
metadata:
  name: cpu1
spec:
  alert_name: CPU1
  group_name: test-group
  formula: |
    node:cluster_cpu_utilization:ratio * 100 > 1
  labels:
    severity: info
  annotations:
    description: |
      This is a test alert
```

Reply:
Sounds good, update to this custom alert and following commands reflect this